node/src/block_ranges.rs

```diff
@@ -443,6 +443,57 @@ impl Sub<&BlockRanges> for BlockRanges {
     }
 }
 
+impl Sub<RangeInclusive<u64>> for BlockRanges {
+    type Output = Self;
+
+    fn sub(mut self, range: RangeInclusive<u64>) -> Self::Output {
+        self.remove_relaxed(range).expect("shouldn't panic");
+        self
+    }
+}
+
+impl SubAssign<RangeInclusive<u64>> for BlockRanges {
+    fn sub_assign(&mut self, range: RangeInclusive<u64>) {
+        self.remove_relaxed(range).expect("shouldn't panic");
+    }
+}
```

Comment:
These can panic if the range is invalid. For example `0..=10` or `11..=10`.

Reply:
Fixed to effectively ignore removal of empty ranges, added tests including this case.